### ISSUE TITLE
didScrollToSegment not being called on programatic segment change

### ISSUE THREAD

**Issue by gerbot150** [patchthecode/JTAppleCalendar#325]
Calling this function:
`self.calendarView.scrollToSegment(.previous, triggerScrollToDateDelegate: true, animateScroll: true)`

On a button action, but the 
```swift
func calendar(_ calendar: JTAppleCalendarView, didScrollToDateSegmentWith visibleDates: DateSegmentInfo) {
``` 
delegate method is not getting called in 6.1.3, so we reverted to 6.1.1 where it is currently working. Not sure if this issue exists on 6.1.2.

Hope this helps someone or can be resolved

**Comment by patchthecode** [comment]
looking into this immediately.
Could be a newly introduced bug.

**Comment by patchthecode** [comment]
Found the issue. 
Will resolve in a several hours.

**Comment by patchthecode** [comment]
The error has been fixed on master branch. 
If you wish to test the latest code with the fix, the [click here](https://github.com/patchthecode/JTAppleCalendar/wiki/Common-Questions)

**Comment by gerbot150** [comment]
Yup, it seems to be working now with the latest code. Thanks for responding and fixing this so quickly!

**Comment by patchthecode** [comment]
Cool. Im the only tester, so if you find any other issue, then let me know. There should be no errors with this library, so I plan to fix them quickly if found. Keep me posted if you find new ones. Cheers 🍻

**Comment by Dean151** [comment]
I just converted my code to Swift 3 and updated JTAppleCalendar to the latest version, and I was about to open this issue if it did not exists !
Cool for the fix. Can't wait for 6.1.4 release then !

**Comment by patchthecode** [comment]
released 6.1.4
closing this issue.

If any new bugs found then let me know. I'll fix.
